src/Core/View.js

```diff
@@ -165,12 +151,6 @@ function _preprocessLayer(view, layer, provider, parentLayer) {
                 providerPreprocessing = Promise.resolve();
```

Comment:
`provider.preprocessDataLayer` should be return Promise to simplify this part.

Reply:
Done, I simplified the whole section of `whenReady`.